src/Aspire.Hosting.Dapr/IDistributedApplicationBuilderExtensions.cs

```diff
@@ -44,13 +44,28 @@ public static IResourceBuilder<IDaprComponentResource> AddDaprComponent(this IDi
     {
         var resource = new DaprComponentResource(name, type) { Options = options };
 
+        builder.Eventing.Subscribe<BeforeStartEvent>((@event, ct) =>
+        {
+            var rns = @event.Services.GetRequiredService<ResourceNotificationService>();
+
+            _ = Task.Run(async () =>
+            {
+                await rns.WaitForDependenciesAsync(resource, ct).ConfigureAwait(false);
+                await rns.PublishUpdateAsync(
+                    resource, s=> s with { State = KnownResourceStates.Running }
+                    ).ConfigureAwait(false);
+            }, ct);
+
+            return Task.CompletedTask;
+        });
+
         return builder
             .AddResource(resource)
             .WithInitialState(new()
             {
                 Properties = [],
                 ResourceType = "DaprComponent",
-                State = KnownResourceStates.Hidden
+                State = KnownResourceStates.Starting
```

Comment:
I think we keep this hidden for now and instead just do copy the waitfor annotations from all components to the side car.

Reply:
Done.